src/main/kotlin/kotlinx/kover/engines/intellij/IntellijAgent.kt

```diff
@@ -55,20 +54,37 @@ internal class IntellijAgent(val config: Configuration) {
             pw.appendLine(generateSmapFile.toString())
             pw.appendLine(smapPath)
             extension.includes.forEach { i ->
-                pw.appendLine(i)
+                pw.appendLine(i.replaceWildcards())
             }
 
             if (extension.excludes.isNotEmpty()) {
                 pw.appendLine("-exclude")
             }
 
-            extension.excludes.forEach { i ->
-                pw.appendLine(i)
+            extension.excludes.forEach { e ->
+                pw.appendLine(e.replaceWildcards())
             }
         }
     }
+
+    private fun String.replaceWildcards(): String {
+        val builder = StringBuilder(length * 2)
```

Comment:
Most of the chars won't result in two chars, so it is logical to place `StringBuilder(length)` here

Reply:
It seems to me that filter are most often used with a wildcard *, or a dot  - in this case an escape symbol will be added, the size will overflow and the capacity will be doubled.